websites/M/Mahara-Tech/presence.ts

```diff
@@ -0,0 +1,188 @@
+type Functionlize<T> = {
+	[P in keyof T]: () => T[P];
+};
+
+interface Route extends Functionlize<Partial<PresenceData>> {
+	path: RegExp;
+	playback?(): boolean;
+	run?(): PresenceData;
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/SjtqzAF.png",
+	PAUSED = "https://raw.githubusercontent.com/PreMiD/Presences/main/.resources/pause.png",
+	PLAYED = "https://raw.githubusercontent.com/PreMiD/Presences/main/.resources/play.png",
+	SEARCHING = "https://raw.githubusercontent.com/PreMiD/Presences/main/.resources/search.png",
+	Discover = "https://raw.githubusercontent.com/PreMiD/Presences/main/.resources/reading.png",
+	VIEWING = "https://raw.githubusercontent.com/PreMiD/Presences/main/.resources/viewing.png",
```

Comment:
These are not actually needed. You can use `Assets.Pause`, `Assets.Play`, etc. even without adding those keys here. (See `index.d.ts`)
```suggestion
```